### PR TITLE
Schedules

### PR DESCRIPTION
**Pull request by kartikeyrajvaidya**

### CODE REVIEW

File: src/components/schemas/fields.yaml [hunk 1]
```diff
@@ -243,3 +243,58 @@ location:
...
+        run_date:
+          nullable: true
+          type: string
+          example: 2021-05-12T10:16:50.410Z
+        
+
+config:
+  type: object
+  nullable: false
+  properties:
```
Comment (jayachandhar) [inline src/components/schemas/fields.yaml:289]: config:
  type: object
  nullable: false
  oneOf:
  - type: object
    properties:
      type: string
      when: <>
  - type: object
     properties:
        type: string
        example: date
        when:
          type: object
          properties:
            run_date:
              nullable: true
              type: string
              example: 2021-05-12T10:16:50.410Z

File: src/fyler/openapi.yaml [hunk 2]
```diff
@@ -65,6 +65,10 @@ tags:
       This API allows you to subscribe to different events.
   - name: Corporate Cards
     description: |
+      This API gives you access to your Cards data.
+  - name: Schedules
+    description: |
+      This API let you create Scheduled jobs.
```
Comment (aditya999123) [inline src/fyler/openapi.yaml:71]: let's

File: src/components/schemas/schedule.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      nullable: true
+      example: report.created
+
+
+schedule_out:
+  type: object
+  properties:
+    id:
+      $ref: './fields.yaml#/id_string'
+    apscheduler_job_id:
```
Comment (aditya999123) [inline src/components/schemas/schedule.yaml:33]: not needed

File: src/components/schemas/schedule.yaml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+      $ref: './fields.yaml#/id_string'
+    created_at:
+      $ref: ./fields.yaml#/created_at
+    updated_at:
+      $ref: ./fields.yaml#/updated_at
+    event_payload:
```
Comment (aditya999123) [inline src/components/schemas/schedule.yaml:39]: not needed

File: src/components/schemas/schedule.yaml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+      $ref: './fields.yaml#/event_payload'
+    config:
+      $ref: './fields.yaml#/config'
+    org_id:
+      $ref: './fields.yaml#/org_id'
+    user_id:
+      $ref: './fields.yaml#/user_id'
+    is_enabled:
+      $ref: ./fields.yaml#/is_enabled
+    event_name:
```
Comment (aditya999123) [inline src/components/schemas/schedule.yaml:49]: can add later

File: src/components/schemas/schedule.yaml [hunk 6]
```diff
@@ -0,0 +1,52 @@
+schedule_in:
+  type: object
+  properties:
+    id:
+      $ref: './fields.yaml#/id_string'
+    apscheduler_job_id:
```
Comment (aditya999123) [inline src/components/schemas/schedule.yaml:6]: same

File: src/components/schemas/fields.yaml [hunk 7]
```diff
@@ -243,3 +243,79 @@ location:
       type: number
       nullable: true
       example: 77.594566
+
+config:
```
Comment (aditya999123) [inline src/components/schemas/fields.yaml:247]: schedule_config

File: src/components/schemas/fields.yaml [hunk 8]
```diff
@@ -243,3 +243,79 @@ location:
...
+      properties:
+        type:
+          nullable: true
+          type: string
+          example: interval
+        when:
+          nullable: true
+          type: object
+          properties:
+            minutes:
```
Comment (aditya999123) [inline src/components/schemas/fields.yaml:260]: can accept other keys here

File: src/components/schemas/fields.yaml [hunk 9]
```diff
@@ -243,3 +243,79 @@ location:
...
+              type: string
+              example: 
+            start_date:
+              nullable: true
+              type: string 
+              example: 2020-11-11 16:29:30.965869
+    - type: object
+      properties:
+        type:
+          nullable: true
```
Comment (aditya999123) [inline src/components/schemas/fields.yaml:275]: not nulla ble in all 3

**Review by aditya999123 — APPROVED** [review]
minor comments

**Pull request closed without merging**